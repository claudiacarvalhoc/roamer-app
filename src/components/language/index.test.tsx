import React from 'react';
import Language, { LanguageProps } from './index';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const getShallowWrapper = (props) => {
    return shallow(<Language {...props} />);
};

describe('<Language />', () => {
    it('render correctly when flag is uk', () => {
        const props: LanguageProps = {
            name: 'English',
            flag: 'uk',
        };
        const component = getShallowWrapper(props);
        expect(toJson(component)).toMatchSnapshot();
    });
    it('render correctly when flag is ru', () => {
        const props: LanguageProps = {
            name: 'Russia',
            flag: 'ru',
        };
        const component = getShallowWrapper(props);
        expect(toJson(component)).toMatchSnapshot();
    });
    it('render correctly when flag is it', () => {
        const props: LanguageProps = {
            name: 'Italian',
            flag: 'it',
        };
        const component = getShallowWrapper(props);
        expect(toJson(component)).toMatchSnapshot();
    });
    it('render correctly when flag is mk', () => {
        const props: LanguageProps = {
            name: 'Macedonian',
            flag: 'mk',
        };
        const component = getShallowWrapper(props);
        expect(toJson(component)).toMatchSnapshot();
    });

});
