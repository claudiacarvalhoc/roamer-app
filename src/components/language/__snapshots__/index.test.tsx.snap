// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Language /> render correctly when flag is it 1`] = `
<div
  className="container"
>
  <ItalyFlag
    className="flag"
    key="vector-{vector}"
  />
  <p
    className="language"
  >
    Italian
  </p>
</div>
`;

exports[`<Language /> render correctly when flag is mk 1`] = `
<div
  className="container"
>
  <MecedonianFlag
    className="flag"
    key="vector-{vector}"
  />
  <p
    className="language"
  >
    Macedonian
  </p>
</div>
`;

exports[`<Language /> render correctly when flag is ru 1`] = `
<div
  className="container"
>
  <RussiaFlag
    className="flag"
    key="vector-{vector}"
  />
  <p
    className="language"
  >
    Russia
  </p>
</div>
`;

exports[`<Language /> render correctly when flag is uk 1`] = `
<div
  className="container"
>
  <UnitedKingdomFlag
    className="flag"
    key="vector-{vector}"
  />
  <p
    className="language"
  >
    English
  </p>
</div>
`;
